client/src/file/locked_keyring.rs

```diff
@@ -0,0 +1,136 @@
+use std::{path::PathBuf, sync::Arc};
+
+#[cfg(feature = "async-std")]
+use async_fs as fs;
+#[cfg(feature = "async-std")]
+use async_lock::{Mutex, RwLock};
+#[cfg(feature = "async-std")]
+use futures_lite::AsyncReadExt;
+#[cfg(feature = "tokio")]
+use tokio::{
+    fs, io,
+    io::AsyncReadExt,
+    sync::{Mutex, RwLock},
+};
+
+use crate::{
+    Secret,
+    file::{Keyring, api, error::Error},
+};
+
+#[derive(Debug)]
+pub struct LockedKeyring {
+    pub(super) keyring: Arc<RwLock<api::Keyring>>,
+    pub(super) path: Option<PathBuf>,
+    pub(super) mtime: Mutex<Option<std::time::SystemTime>>,
+}
+
+impl LockedKeyring {
+    /// Unlocks a keyring and validates it
+    pub async fn unlock(self, secret: Secret) -> Result<Keyring, Error> {
```

Comment:
Given that we also have a Keyring inside `api::Keyring`, maybe we should rename this one to UnlockedKeyring.

Reply:
Renamed in the latest commits.